velox/functions/sparksql/aggregates/tests/ModeAggregateTest.cpp

```diff
@@ -0,0 +1,408 @@
+       makeFlatVector<Timestamp>(
+           {Timestamp{0, 17'123'456},
+            Timestamp{1, 17'123'456},
+            Timestamp{1, 17'123'456},
+            Timestamp{0, 17'123'456}})});
+
+  testMode("mode(c1)", {"c0"}, vector1, vector2, expected);
+}
+
+TEST_F(ModeAggregateTest, groupByDate) {
+  vector_size_t num = 10;
+
+  auto vector1 = makeFlatVector<int32_t>(
+      num, [](vector_size_t row) { return row % 3; }, nullEvery(4));
+  auto vector2 = makeFlatVector<int32_t>(
+      num, [](vector_size_t row) { return row % 2; }, nullEvery(5), DATE());
+
+  auto expected = makeRowVector(
+      {makeNullableFlatVector<int32_t>({std::nullopt, 0, 1, 2}),
+       makeFlatVector<int32_t>({0, 1, 1, 0}, DATE())});
+
+  testMode("mode(c1)", {"c0"}, vector1, vector2, expected);
+}
+
+TEST_F(ModeAggregateTest, groupByInterval) {
+  vector_size_t num = 30;
+
+  auto vector1 = makeFlatVector<int32_t>(num, [](auto row) { return row; });
+  auto vector2 = makeFlatVector<int64_t>(
+      num,
+      [](auto row) { return row % 5 == 0 ? 2 : row + 1; },
+      nullEvery(5),
+      INTERVAL_DAY_TIME());
+
+  testModeWithDuck(vector1, vector2);
+}
+
+TEST_F(ModeAggregateTest, groupByString) {
+  std::vector<std::string> strings = {
+      "grapes",
+      "oranges",
+      "sweet fruits: apple",
+      "sweet fruits: banana",
+      "sweet fruits: papaya",
+  };
+
+  auto keys = makeFlatVector<int16_t>(
+      1'002, [](auto row) { return row % 5; }, nullEvery(5));
+  auto data = makeFlatVector<StringView>(
+      1'002,
+      [&](auto row) { return StringView(strings[row % strings.size()]); },
+      nullEvery(5));
+  testModeWithDuck(keys, data);
+}
+
+TEST_F(ModeAggregateTest, globalInteger) {
+  vector_size_t num = 30;
+  auto vector = makeFlatVector<int32_t>(
+      num, [](vector_size_t row) { return row % 7; }, nullEvery(7));
+
+  testGlobalModeWithDuck(vector);
+}
+
+TEST_F(ModeAggregateTest, globalDecimal) {
+  vector_size_t num = 10;
+  auto longDecimalType = DECIMAL(20, 2);
+  auto vector1 = makeFlatVector<int128_t>(
+      num,
+      [](vector_size_t row) { return row % 4; },
+      nullEvery(7),
+      longDecimalType);
+
+  auto expected1 = makeRowVector(
+      {makeFlatVector<int128_t>(std::vector<int128_t>{1}, longDecimalType)});
+
+  testMode("mode(c1)", {}, vector1, vector1, expected1);
+
+  auto shortDecimalType = DECIMAL(6, 2);
+  auto vector2 = makeFlatVector<int64_t>(
+      num,
+      [](vector_size_t row) { return row % 4; },
+      nullEvery(7),
+      shortDecimalType);
+
+  auto expected2 = makeRowVector(
+      {makeFlatVector<int64_t>(std::vector<int64_t>{1}, shortDecimalType)});
+
+  testMode("mode(c1)", {}, vector2, vector2, expected2);
+}
+
+TEST_F(ModeAggregateTest, globalUnknown) {
+  auto vector = makeAllNullFlatVector<UnknownValue>(6);
+
+  auto expected = makeRowVector({
+      BaseVector::createNullConstant(UNKNOWN(), 1, pool()),
+  });
+
+  testMode("mode(c1)", {}, vector, vector, expected);
```

Comment:
The expected value should be
```
spark-sql (default)>  SELECT mode(x) FROM (VALUES(null), (null), (null), (null), (null), (null)) AS t(x);
NULL
```

Reply:
You could take this as example, test the unknown type in complex type.
https://github.com/facebookincubator/velox/pull/5408/files